Re #2361: fix seg combobox filling

Punzo opened the pull request:
Fix https://github.com/OHIF/Viewers/issues/2361

dataset:
http://localhost:3000/projects/idc-dev-etl/locations/us-central1/datasets/idc_tcia_dev/dicomStores/idc_tcia/study/1.3.6.1.4.1.14519.5.2.1.7695.2311.111788911924319643511738623890

igoroctaviano commented on extensions/dicom-segmentation/src/getSourceDisplaySet.js:
```diff
@@ -36,14 +36,32 @@ const _getReferencedDisplaySet = (segDisplaySet, studies) => {
+  }
+
+  if (metadata.ReferencedImageSequence &&
+    (!referencedSeriesInstanceUIDs || referencedSeriesInstanceUIDs.length === 0)) {
```
Can we extract this block into a separate jsDocs function? this will help understand this code block in the future

igoroctaviano commented:
> http://localhost:3000/projects/idc-dev-etl/locations/us-central1/datasets/idc_tcia_dev/dicomStores/idc_tcia/study/1.3.6.1.4.1.14519.5.2.1.7695.2311.111788911924319643511738623890

2 issues:
- When selecting the second segmentation and going back to the first one the dropdown doesn't update.
- Invalid date in dropdown item.

Punzo commented:
> > http://localhost:3000/projects/idc-dev-etl/locations/us-central1/datasets/idc_tcia_dev/dicomStores/idc_tcia/study/1.3.6.1.4.1.14519.5.2.1.7695.2311.111788911924319643511738623890
> 
> 2 issues:
> 
> * When selecting the second segmentation and going back to the first one the dropdown doesn't update.


Fixed 

> * Invalid date in dropdown item.

This is not a bug, the metadata are missing the dates.

I have updated the PR, please review again, thanks!

igoroctaviano commented:
Just retested, found some issues...
- Invalid date:
![Screen Shot 2021-04-20 at 16 56 37](https://user-images.githubusercontent.com/13886933/115456283-60545400-a1f9-11eb-921c-82cd72ced98f.png)
- Segmentation option with no segments when selecting a segmentation in the dropdown and then clicking other series with no segmentation:
![Screen Shot 2021-04-20 at 16 59 11](https://user-images.githubusercontent.com/13886933/115456606-bc1edd00-a1f9-11eb-877a-bd756a32c4c6.png)

Punzo commented:
> Just retested, found some issues...
> 
> * Invalid date:
>   ![Screen Shot 2021-04-20 at 16 56 37](https://user-images.githubusercontent.com/13886933/115456283-60545400-a1f9-11eb-921c-82cd72ced98f.png)

This is not a bug  (see also https://github.com/OHIF/Viewers/pull/2369#issuecomment-823321797), all the segmentation series of this study miss the series time and date. So returning Invalid date is right.


> * Segmentation option with no segments when selecting a segmentation in the dropdown and then clicking other series with no segmentation:
>   ![Screen Shot 2021-04-20 at 16 59 11](https://user-images.githubusercontent.com/13886933/115456606-bc1edd00-a1f9-11eb-877a-bd756a32c4c6.png)

This is a bug in the react select element, see https://github.com/JedWatson/react-select/issues/3066. I.e. undefined values do not clear the combobox, but only null. I fixed it in by forcing null if undefined is detected:
Fixed in https://github.com/OHIF/Viewers/pull/2369/commits/27f2f0bc7c551f34570dad4765083b29105410cf

igoroctaviano reviewed (approved):
LGTM!
